File: kotlin/Adapter.kt
class InternalClass {
    fun fetch() = "get user info"
}

class ExternalClass {
    fun search() = "get user info"
}

class Adapter(private val external: ExternalClass) {
    fun fetch(): String = external.search()
}

fun main() {
    val ext = ExternalClass()
    val adapter = Adapter(ext)
    println(adapter.fetch())
}

File: cpp/adapter.cpp
#include <iostream>
using namespace std;

class InternalClass {
public:
    string fetch() { return "get user info"; }
};

class ExternalClass {
public:
    string search() { return "get user info"; }
};

class Adapter {
    ExternalClass* external;

public:
    Adapter(ExternalClass* ext) : external(ext) {}
    string fetch() { return external->search(); }
};

int main() {
    ExternalClass ext;
    Adapter adapter(&ext);
    cout << adapter.fetch() << endl;
    return 0;
}